src/Nerdbank.MessagePack/Converters/ArrayConverter`1.cs

```diff
@@ -28,7 +28,14 @@ internal class ArrayConverter<TElement>(MessagePackConverter<TElement> elementCo
 		TElement[] array = new TElement[count];
 		for (int i = 0; i < count; i++)
 		{
-			array[i] = elementConverter.Read(ref reader, context)!;
+			try
```

Comment:
Instead of a try/catch block within the loop, define it outside the loop, and define the iterator variable outside the loop as well so that it can be accessed from the `catch` block.
Review your other changes and apply the same policy to avoid try's inside loops. This is for perf reasons.